src/server/lib/oauth/client.js

```diff
@@ -131,6 +131,9 @@ async function getOAuth2AccessToken (code, provider, codeVerifier) {
     { algorithm: 'ES256', keyid: keyId }
     )
     params.client_secret = clientSecret
+  } else if (provider.id === 'wechat') {
+    params.appid = provider.clientId || provider.appid
+    params.secret = provider.clientSecret;
```

Comment:
Custom behavior like this won't be supported in the upcoming major release. Read #2411 and the corresponding RFC #1846 for context.